### ISSUE TITLE
CurrencySymbol wrong in .Net ^ 

### ISSUE THREAD

**Issue by narcis-ro** [dotnet/runtime#76437]
### Description

CurrencySymbol for culture "ro-RO" has changed between .Net Framework and .Net 6. The correct symbol is `lei` (as it was in .net 4.8)

### Reproduction Steps

```
            Console.OutputEncoding = System.Text.Encoding.UTF8;

            CultureInfo.CurrentCulture = new CultureInfo("ro-RO");
            CultureInfo.CurrentUICulture = new CultureInfo("ro-RO");

            CultureInfo culture1 = CultureInfo.CurrentCulture;
            Console.WriteLine("The current culture is {0} -> {1}", culture1.Name, culture1.DisplayName);
            CultureInfo uiCulture2 = CultureInfo.CurrentUICulture;
            Console.WriteLine("The current UI culture is {0} -> {1}\n", uiCulture2.Name, culture1.DisplayName);

            Console.WriteLine();

            RegionInfo myRI1 = new RegionInfo("RO");
            Console.WriteLine("   Name:                         {0}", myRI1.Name);
            Console.WriteLine("   DisplayName:                  {0}", myRI1.DisplayName);
            Console.WriteLine("   EnglishName:                  {0}", myRI1.EnglishName);
            Console.WriteLine("   IsMetric:                     {0}", myRI1.IsMetric);
            Console.WriteLine("   ThreeLetterISORegionName:     {0}", myRI1.ThreeLetterISORegionName);
            Console.WriteLine("   ThreeLetterWindowsRegionName: {0}", myRI1.ThreeLetterWindowsRegionName);
            Console.WriteLine("   TwoLetterISORegionName:       {0}", myRI1.TwoLetterISORegionName);
            Console.WriteLine("   CurrencySymbol:               {0}", myRI1.CurrencySymbol);
            Console.WriteLine("   ISOCurrencySymbol:            {0}", myRI1.ISOCurrencySymbol);

            Console.ReadLine();
```

### Expected behavior

CurrencySymbol is "lei"

![image](https://user-images.githubusercontent.com/51541214/193252657-9e242667-06bd-4ce9-a747-ea843a08cc0b.png)

### Actual behavior

CurrencySymbol is "RON"

![image](https://user-images.githubusercontent.com/51541214/193252690-1fc5299e-e90d-4c20-8073-ee10e0d2938c.png)

### Regression?

Regression from .net 4.8

### Configuration

Win10 - x64
.net core sdk 6.0.400

**Comment by Clockwork-Muse** [comment]
I'm currently physically in Romania, and was wondering why I was seeing two different sets of currency designations....

Anyways, the actual issue here is (almost certainly) that [.NET switched over to ICU on Windows](https://learn.microsoft.com/en-us/dotnet/core/extensions/globalization-icu) - there's no bug in .NET per se (and wouldn't have been before this either, because this sort of data was maintained by the Windows OS).
You can try switching to use the legacy NLS as documented (at least to verify the source of the issue), but the recommendation is to migrate to use the ICU source.  Note that there was never a guarantee that the NLS would always return the same data in the first place.

You could potentially report this to the ICU maintainers as a data bug.

**Comment by tarekgh** [comment]
@Clockwork-Muse is correct. CLDR Unicode standard is defining the currency symbol as [RON](https://github.com/unicode-org/cldr/blob/d70f5d0d6ddc3f68899c3e8f7bce2569370160d1/common/main/ro.xml#L7740). You can open a [CLDR Ticket](https://unicode-org.atlassian.net/jira/software/c/projects/CLDR/issues/) for that you are sure the currency symbol is wrong. You can switch to [NLS mode](https://learn.microsoft.com/en-us/dotnet/core/extensions/globalization-icu#use-nls-instead-of-icu) if needed too.

**Comment by narcis-ro** [comment]
@tarekgh the bug is that it doesn’t read the narrow symbol.

extract from the link you provided:

```

<currency type="RON">
				<displayName>leu românesc</displayName>
				<displayName count="one">leu românesc</displayName>
				<displayName count="few">lei românești</displayName>
				<displayName count="other">lei românești</displayName>
				<symbol>RON</symbol>
				<symbol alt="narrow">lei</symbol>
			</currency>

```

ISOCurrencySymbol would be RON and CurrencySymbol would be the narrow version


Check american dolar for example:
```
<currency type="USD">
				<displayName>dolar american</displayName>
				<displayName count="one">dolar american</displayName>
				<displayName count="few">dolari americani</displayName>
				<displayName count="other">dolari americani</displayName>
				<symbol>USD</symbol>
				<symbol alt="narrow">$</symbol>
			</currency>
```

**Comment by tarekgh** [comment]
@narcis-ro you are right about the data, but .NET is not reading the data directly. Instead, we use ICU APIs to read the data. We are calling the ICU with the correct information and that is what is returning `RON`.  You look at [ICU Locale “Romanian (Romania)” (ro_RO)](https://www.localeplanet.com/icu/ro-RO/index.html) which shows the same result. Also looking at [Supported Currencies (ICU)](https://help.salesforce.com/s/articleView?id=sf.admin_supported_currencies.htm&type=5) which also shows the numbers formatted with `RON` for Romanian culture:

```
Romanian (Romania)ro_RO | Leu românesc | RON | 1.234.567,57 RON-1.234.567,57 RON
```

It may be worth [opening the ticket to ICU](https://unicode-org.atlassian.net/jira/software/c/projects/ICU/issues/) instead of CLDR. and here is the ICU calling code showing the problem:

```C
    UErrorCode status = U_ZERO_ERROR;
    UChar value[100];

    UNumberFormat* pFormat = unum_open(UNUM_DECIMAL, NULL, 0, "ro_RO", NULL, &status);
    int32_t l = unum_getSymbol(pFormat, UNUM_CURRENCY_SYMBOL, value, 100, &status);
    printf("Currency Symbole: %ls \n", value);   <<<<< this print RON 
    unum_close(pFormat);

```

**Comment by tarekgh** [comment]
I'll look more here to see if there is a way, we can read the narrow currency symbol from ICU. I'll try to report back here. But in general, we need to confirm which symbol is correct for `ro_RO` as it looks both are correct. otherwise, `RON` wouldn't be listed as a symbol.

**Comment by narcis-ro** [comment]
Before operning the ticket I did find this ICU issue reported 4 years ago:
https://unicode-org.atlassian.net/plugins/servlet/mobile?originPath=%2Fbrowse%2FCLDR-9464#issue/CLDR-9464

It seems at that time the narrow symbol was missing indeed. In the meantime it was added to both ICU and CLDR

what is weird thst ICU for c++ does read it correctly.

https://github.com/unicode-org/icu/blob/dcd19ae9bc0914f8be439a2de1a5fbc17e7b5447/icu4c/source/data/curr/ro.txt

The root is the same:

https://github.com/unicode-org/icu/blob/64b35481263ac4df37a28a9c549553ecc9710db2/icu4c/source/data/curr/root.txt

The [link](https://www.localeplanet.com/icu/ro-RO/index.html) to localeplanet displays wrong info too. Same as the link to sales force. 
As native Romanian, I can guarantee 100% that “lei” is the narrow symbol, and used in the country for any day to day use.
https://st.unicode.org/cldr-apps/v#/ro/C_SEEurope/77b33e6ae2756b1b Shows correct info.

**Comment by tarekgh** [comment]
The question that I don't have a clear answer for is, why does CLDR list `RON` as a currency symbol if it is not correct? Is this means it is correct to use it too as currency symbol? I trust you `lei` is correct, so why not just use it as currency symbol and remove `RON`? 

You have pointed at American dollar example, but this snippet was not from the `en` locale file, look at the data from `en` file and it is clear there and not listing `$` as narrow symbol. Instead, it is just the currency symbol. 

```Xml
			<currency type="USD">
				<displayName>US Dollar</displayName>
				<displayName count="one">US dollar</displayName>
				<displayName count="other">US dollars</displayName>
				<symbol>$</symbol>
			</currency>
```
https://github.com/unicode-org/cldr/blob/d70f5d0d6ddc3f68899c3e8f7bce2569370160d1/common/main/en.xml#L5946

So, I expect the same for Romanian symbol to list `lei` as the currency symbol and not just the narrow one. Also, having ICU returning `RON` as a currency symbol for `ro_RO` looks incorrect if `lei` should be the one. right?

**Comment by narcis-ro** [comment]
``

**Comment by narcis-ro** [comment]
When you [click](https://st.unicode.org/cldr-apps/v#/ro/C_SEEurope/3f063922920fd8)  on the narrow form it gives you the following explanation.


```
The NARROW form of the symbol used for the currency with the ISO currency code = RON, when the known context is already enough to distinguish the symbol from other currencies that may use the same symbol. Normally, this does not need to be changed from the inherited value. For more information, see https://cldr.unicode.org/translation/currency-names-and-symbols.
```

Following that [link](https://cldr.unicode.org/translation/currency-names-and-symbols) to the rules of symbols:


```
If a symbol is not widely recognized around the world (eg shekel ₪)
Where the currency is official in a country, use that symbol in locales with that country (eg IL)
Where the currency would be widely recognized by users of a language, use it in the base language locale (eg he/iw).
```
For use of **alt** is detailed in that link [currency process](https://cldr.unicode.org/index/cldr-spec/currency-process) 

So I guess the bug for .net is notreading the **alt** aka **narrow** symbol when avaiable for a culture

**Comment by tarekgh** [comment]
@narcis-ro thanks for the details. This is helpful. I am reopening this issue.

**Comment by narcis-ro** [comment]
@tarekgh glad I could be of any help.
